### PR TITLE
Fix issues, ready for review for v1.0

### PR DESCRIPTION
**Pull request by Timuuc**
FIx issues #4  #5  #6 #7 #8

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -240,7 +240,7 @@ to add a cmd module:
         RET_ERROR_FILE_OPEN_FAILED       = 6
         RET_ERROR_NO_USERINFORMATION     = 7
         RET_ERROR_MISSING_UNSERINFO      = 8
-        RET_ERROR_MISSING_LOGIN_INFO     = 9
+        RET_ERROR_MISSING_ARG_INFO     = 9
```
Comment (BlueAndi) [inline README.md:243]: Indention is different than the others.

File: README.md [hunk 2]
```diff
@@ -258,7 +258,7 @@ to add a cmd module:
                                                "or stored information file",
         Ret.RET_ERROR_MISSING_UNSERINFO      : "both -user and -pw option must be provided " + 
                                                "to store useriformation",
-        Ret.RET_ERROR_MISSING_LOGIN_INFO     : "At least one of the options must be" + 
+        Ret.RET_ERROR_MISSING_ARG_INFO     : "At least one of the options must be" +
```
Comment (BlueAndi) [inline README.md:261]: Indention is different than the others.

File: src/pyJiraCli/__main__.py [hunk 3]
```diff
@@ -74,10 +72,15 @@
 # Functions
 ################################################################################
 def add_parser():
-    """"add parser for command line arguments
-    
-        return the parser after all the modules have been registered"""
-
+    """ add parser for command line arguments
```
Comment (BlueAndi) [inline src/pyJiraCli/__main__.py:75]: Sentence form ... start uppercase ... use dot, etc.

File: src/pyJiraCli/__main__.py [hunk 4]
```diff
@@ -74,10 +72,15 @@
...
+        set the execute function of each 
+        cmd module as callback for the subparser command
+        return the parser after all the modules have been registered
+        and added their subparsers
+        
+    Returns:
+        parser  the parser for commandline arguments
```
Comment (BlueAndi) [inline src/pyJiraCli/__main__.py:82]: After the type is a ':' missing.

File: src/pyJiraCli/cmd_export.py [hunk 5]
```diff
@@ -87,50 +85,75 @@ def register(subparser):
 
     return sb_export
 
-def execute(args):
-    """execute command function"""
+def execute(args) -> Ret:
+    """ Execute the export command function.
+    
+    Args: 
+        args        the command line arguments
```
Comment (BlueAndi) [inline src/pyJiraCli/cmd_export.py:92]: Type is missing ... use sentence form ...

File: src/pyJiraCli/cmd_export.py [hunk 6]
```diff
@@ -87,50 +85,75 @@ def register(subparser):
...
-       path:  path to the folder where the file shall be stored
-       csv:   flag, if true save the file in csv format
+    Args:
+        issue (str): issue key (used as filename if no name or file provided)
+        file (str):  the filename for the file which will be created
+        path (str):  path to the folder where the file shall be stored
+        csv (bool):  flag, if true save the file in csv format
+
+    Returns:
+        str:   path where the ticket file will be stored or None
```
Comment (BlueAndi) [inline src/pyJiraCli/cmd_export.py:151]: For tuple use: tuple[str, Ret]

File: src/pyJiraCli/cmd_export.py [hunk 7]
```diff
@@ -159,54 +182,53 @@ def _get_filepath(issue, file, path, csv):
...
         else:
-            return None, Ret.RET_ERROR_FILE_NOT_FOUND
+            ret_status = Ret.RET_ERROR_FILE_NOT_FOUND
 
-    return file_path, Ret.RET_OK
+    return file_path, ret_status
 
-def _export_ticket_to_file(issue_key, filepath, user, pw, csv):
-    """"export jira issue from server to json or csv file
+def _export_ticket_to_file(issue_key:str, filepath:str, user:str, pw:str, csv:bool) -> Ret:
```
Comment (BlueAndi) [inline src/pyJiraCli/cmd_export.py:197]: If you specify the type for the return value, it should be done for all to be consistent.
Its up to you ...

File: src/pyJiraCli/cmd_import.py [hunk 8]
```diff
@@ -54,15 +56,14 @@
...
-        subparser: subparser
+    """ Register subparser commands for the import module.
+        
+    Args:
+        subparser (obj):  the command subparser provided via __main__.py
         
-        return:
-        commmand parser of this module
+    Returns:
+        obj:  the commmand parser of this module
```
Comment (BlueAndi) [inline src/pyJiraCli/cmd_import.py:65]: Sentence ... :-)

File: src/pyJiraCli/cmd_login.py [hunk 9]
```diff
@@ -53,13 +54,13 @@
...
         
-        param:
-        subparser: subparser
+    Args:
+        subparser (obj):   the command subparser provided via __main__.py
         
-        return:
-        the cmd parser for this module
+    Returns:
+        obj:    the commmand parser of this module
```
Comment (BlueAndi) [inline src/pyJiraCli/cmd_login.py:63]: Sentence ... this comment is valid for all modules. Please check them agian.

File: src/pyJiraCli/data_type.py [hunk 10]
```diff
@@ -0,0 +1,54 @@
+"""The main module with the program entry point."""
```
Comment (BlueAndi) [inline src/pyJiraCli/data_type.py:1]: Wrong description.

File: src/pyJiraCli/file_handler.py [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+
+        Returns:
+            Ret:   Ret.RET_OK if succesfull, corresponding error code if not
+        """
+        ret_status = Ret.RET_OK
+
+        try:
+            self._file = open(self._path, mode=file_mode, encoding='utf-8') # pylint: disable=consider-using-with
+
+        except (OSError, IOError) as e:
```
Comment (BlueAndi) [inline src/pyJiraCli/file_handler.py:163]: File not found error is missing.

File: src/pyJiraCli/file_handler.py [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+
+from pyJiraCli.ret import Ret
+################################################################################
+# Variables
+################################################################################
+FILE_ATTRIBUTE_HIDDEN = 0x02
+################################################################################
+# Classes
+################################################################################
+class FileHandler:
```
Comment (BlueAndi) [inline src/pyJiraCli/file_handler.py:46]: What is the advantage of the file handler? The methods look like they just hide the standard file api.

File: src/pyJiraCli/cmd_print.py [hunk 13]
```diff
@@ -58,10 +65,29 @@ def register(subparser):
 
     return sb_search
 
-def execute(args):
-    """execute command function"""
+def execute(args) -> Ret:
```
Comment (BlueAndi) [inline src/pyJiraCli/cmd_print.py:68]: Return type ... consistent please.

File: src/pyJiraCli/cmd_print.py [hunk 14]
```diff
@@ -58,10 +65,29 @@ def register(subparser):
...
+        args (obj): the command line arguments
+        
+    Returns:
+        Ret:   Ret.RET_OK if succesfull, corresponding error code if not
+    """
     return _cmd_print(args.issue, args.user, args.pw)
 
-def _cmd_print(issue, user, pw):
-    """print ticket information from jira issue to console"""
+def _cmd_print(issue:str, user:str, pw:str) -> Ret:
```
Comment (BlueAndi) [inline src/pyJiraCli/cmd_print.py:79]: Return type ... consistent please.

**Review by BlueAndi — CHANGES_REQUESTED** [review]

File: src/pyJiraCli/file_handler.py (hunk 12 above)
Comment (Timuuc) [reply src/pyJiraCli/file_handler.py:46]: Remove duplicate code with single try-catch block for file opening

**Review by BlueAndi — APPROVED** [review]

**Pull request merged**